### ISSUE TITLE
Tor won't start after installing GlobaLeaks

### ISSUE THREAD

**Issue by germfue** [globaleaks/GlobaLeaks#1948]
**Current behavior**

A fresh installation of GlobaLeaks will change the apparmor configuration for tor. In `globaleaks.postinst`, we write the file `/etc/apparmor.d/local/system_tor` but we only grant write permissions for the folder: `/var/globaleaks/torhs/`
See `globaleaks.postinst`
```
echo "/var/globaleaks/torhs/ w," >> /etc/apparmor.d/local/system_tor
echo "/var/globaleaks/torhs/** rwk," >> /etc/apparmor.d/local/system_tor
```

Problem is that by allowing write permissions only for `/var/globaleaks/torhs/`, tor service fails to start with an error (in `/var/log/syslog`):
```
Apr 20 18:04:20 guest tor[9308]: Apr 20 18:04:20.075 [warn] Directory /var/globaleaks/torhs/ cannot be read: Permission denied
Apr 20 18:04:20 guest tor[9308]: Apr 20 18:04:20.075 [warn] Checking service directory /var/globaleaks/torhs/ failed.
Apr 20 18:04:20 guest tor[9308]: Apr 20 18:04:20.075 [warn] Failed to parse/validate config: Failed to configure rendezvous options. See logs for details.
Apr 20 18:04:20 guest tor[9308]: Apr 20 18:04:20.075 [err] Reading config failed--see warnings above.
Apr 20 18:04:20 guest kernel: [  242.520033] audit: type=1400 audit(1492711460.070:47): apparmor="DENIED" operation="open" profile="system_tor" name="/var/globaleaks/torhs/" pid=9308 comm="tor" requested_mask="r" denied_mask="r" fsuid=0 ouid=112
Apr 20 18:04:20 guest systemd[1]: tor@default.service: Main process exited, code=exited, status=1/FAILURE
```

**Expected behavior**

Tor should initialize without any error

**Steps to reproduce the problem or feature illustration**

1. Modify `/etc/apparmor.d/local/system_tor` to contain
`/var/globaleaks/torhs/ w,`
2. restart tor (service tor restart)

I add here the diff showing the change that would fix it:
```
$ git diff globaleaks.postinst
diff --git a/debian/globaleaks.postinst b/debian/globaleaks.postinst
index 8178ef6..70fc3b0 100755
--- a/debian/globaleaks.postinst
+++ b/debian/globaleaks.postinst
@@ -44,7 +44,7 @@ if ! id -u globaleaks >/dev/null 2>&1; then
 
   # Add AppArmor rules for allowing tor to access to directory /var/globaleaks/torhs/
   if ! $(grep -q -i globaleaks /etc/apparmor.d/local/system_tor); then
-    echo "/var/globaleaks/torhs/ w," >> /etc/apparmor.d/local/system_tor
+    echo "/var/globaleaks/torhs/ rw," >> /etc/apparmor.d/local/system_tor
     echo "/var/globaleaks/torhs/** rwk," >> /etc/apparmor.d/local/system_tor
 
     # restart apparmor service in order to load tor rules
```
**What is the motivation or use case for changing the behavior?**

tor should work out of the box

**GlobaLeaks version:**

v2.67.4

**Browser:**

- [] All browsers
- [ ] Chrome:  v45.0.1
- [ ] Firefox: v55.0.1
- [ ] Safari:
- [ ] Mobile Chrome:
- [ ] Other: 
- [x] Not Applicable

**Server Operating System and Version (if applicable):**
 
Ubuntu 16.04 & 14.04

**Client Operating System and Version (if applicable):**

NA

**Comment by NSkelsey** [comment]
Hi @germfue thanks for reporting this issue and producing the ticket.

This is an interesting denial that app_armor is producing. That configuration for app_armor has functioned for the [last 4 years](https://github.com/globaleaks/GlobaLeaks/blame/fae456d834f295d38cbef8dd447eeb60611d82b8/debian/postinst).

I just tried the install with globaleaks v2.67.4, tor v0.2.7.6 and ubuntu 16.04. The setup and launch worked fine. I am not sure what is different between are environments. Which version of Tor and Ubuntu did you witness the failure on?

**Comment by germfue** [comment]
hi @NSkelsey,
Sorry for the delay in my answer, I missed the alert after your comment.
I found the error using Vultr with the following software:
- tor: 0.2.9.10-1~xenial+1
- Ubuntu: 
```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
```

**Comment by evilaliv3** [comment]
Dear @germfue is this issue solved for you? Do you have relevant information that you could share for other users?

**Comment by fpietrosanti** [comment]
Other Tor/Apparmor related error into the forum https://forum.globaleaks.org/t/after-upgrade-from-14-04-to-16-04-onion-site-doesnt-work/97/6

**Comment by evilaliv3** [comment]
Fixes related to this issue exists and is going to be shipped in 2.67.9.

The could have been related to both apparmor and systemd causing a failure in the miggle of ```globaleaks.postinst```

To fix it it should be executed manually each command present in https://github.com/globaleaks/GlobaLeaks/blob/master/debian/globaleaks.postinst that has not been executed due to a failure.

No current solution exists.
